### PR TITLE
(EWL-939): Player interaction wrapper 📺

### PR DESCRIPTION
**Pull request by JulienKode**
# (EWL-939): Player interaction wrapper 📺

## Checklist

- Browser compatibility checked:
  - [x] Safari
  - [x] Chrome
  - [x] Firefox
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Description

- [x] add PlayerWrapper witch handle user events, like some keyboard event, mouse event, touch event on mobile screen etc..
- [x] create a hooks to handle the display logic it will make the choice to make the UI controls active or not 
- [x] add hooks testing library
- [x] add demo animation when we will show the player to the end user ( transition may change between the position of the bottons )

## Test Plan

Added tests cases for both features 
Tested with https://github.com/EurosportDigital/eurosport-toolkit-player/pull/53 the from eurosport-toolkit-player

# Test plan

- [x] Empty data
- [x] Invalid data
- [x] Correct data

# Checklist
### Before code review

- [x] Move Jira ticket into PR
- [x] Check with another developers that features sounds good

### After code review

- [ ] Move Jira ticket into QA

### CODE REVIEW

File: src/components/JWPlayerETP/PlayerWrapper/index.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import useInteraction from './useInteraction';
+
+const PlayerWrapper = ({ children, isPlaying }) => {
+  const { onKeyUp, active, handlePlayerInteraction } = useInteraction({ isPlaying });
+  return (
+    // eslint-disable-next-line jsx-a11y/no-static-element-interactions
```
[the remark this answers is not in the document]
Comment (JulienKode) [reply src/components/JWPlayerETP/PlayerWrapper/index.js]: I've try but didn't succeed, he wants us to set a `role` for that div ... 
I didn't find any role corresponding to the job of that div: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles

Moreover JWPlayer already set a `role` for the video div

**Comment by JulienKode** [comment]
@saad-kh We can test the interaction without the UI that is coming as it's about showing the skin when we needs to

File: src/components/JWPlayerETP/PlayerWrapper/index.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (JulienKode) [reply src/components/JWPlayerETP/PlayerWrapper/index.js]: You're right I'll rename it `PlayerSkin`
Controls are coming in #288 

With @leozero we see that for the moment we will keep it inside the Player folder

[the remark this answers is not in the document]
Comment (JulienKode) [reply src/components/JWPlayerETP/PlayerWrapper/index.js]: You're right I'll rename it `PlayerSkin`
Controls are coming in #288 

With @leozero we see that for the moment we will keep it inside the Player folder

File: src/components/JWPlayerETP/PlayerWrapper/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (JulienKode) [reply src/components/JWPlayerETP/PlayerWrapper/index.js]: 7de47088946e989e102032fa5b3677f5b33d904e should solve the problem

File: src/components/JWPlayerETP/PlayerWrapper/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (JulienKode) [reply src/components/JWPlayerETP/PlayerWrapper/index.js]: 7de47088946e989e102032fa5b3677f5b33d904e should solve it

File: src/components/JWPlayerETP/PlayerWrapper/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (JulienKode) [reply src/components/JWPlayerETP/PlayerWrapper/index.js]: I succeed to remove it by using styled component

[the remark this answers is not in the document]
Comment (JulienKode) [reply src/components/JWPlayerETP/PlayerWrapper/index.js]: What did you think ?

**Comment by saad-kh** [comment]
I mean we're doing blind dev now, I think it's better to have a bare bone wrapper where clicking on the wrapper trigger the start and pause of the video. Once this behavior and code is validated we can start reason about what to do next.
Now when I run storybook I have no clue about what it's going on.

**Comment by JulienKode** [comment]
Mouse interaction to show the skin are not like controls interaction.
Here it's just about showing the UI when the mouse is over the player

**Pull request merged**